grapheneapi/graphenews.py

```diff
@@ -179,16 +181,21 @@ def run_forever(self) :
                     coro = loop.create_connection(self.factory, self.host,
                                                   self.port, ssl=context)
                 else :
+
                     coro = loop.create_connection(self.factory, self.host,
                                                   self.port, ssl=self.ssl)
-
                 loop.run_until_complete(coro)
                 loop.run_forever()
             except KeyboardInterrupt:
                 break
 
-            print("Trying to re-connect in 10 seconds!")
-            time.sleep(10)
-
+            try:
+                print("Multiple witnesses configured, trying another witness.")
+                GrapheneWebsocketRPC.__init__(self, self.url_list, self.username, self.password)
+                self.ssl, self.host, self.port, self.resource, self.path, self.params = parseWsUrl(self.url)
+                self.connect()
```

Comment:
loop.run_fovever() is in a try/except block, if it raises an exception it goes to the 2nd try except block in the while loop, after that code is executed it goes back to the first try/except block which will block until run_forever raises an exception.

The reconnect is already in the while loop, am I misunderstanding something here?